### PR TITLE
DBZ-6262 Enable pass through configurations for sync topic and rebalancing topic

### PR DESCRIPTION
**Pull request by ShuranZhang**
Enable the user to configure arbitrary configurations for sync topic with prefix "connector.spanner.sync.topic.config.*" and  for rebalancing topic with prefix "connector.spanner.rebalancing.topic.config.*".

Context: https://debezium.zulipchat.com/#narrow/stream/302529-users/topic/High.20CPU.20consumption.20in.20Spanner.20debezium/near/344217302

For example, this pr will enable users to configure the compression type for sync topic and rebalancing topic in config JSON file by setting:  
~~~~
"connector.spanner.sync.topic.config.compression.type" : "gzip",
"connector.spanner.rebalancing.topic.config.compression.type":"snappy",
~~~~

Tested:
Screenshot 2023-04-05 at 11 37 38 AM
Screenshot 2023-04-05 at 11 38 01 AM

### CODE REVIEW

**Review by jpechane — CHANGES_REQUESTED** [review]
Please extend the scope of the issue to cover all topics and also all producer and consumer factories. It is not going to be a lot of code yet it will be very useful for the user.

File: src/main/java/io/debezium/connector/spanner/db/dao/SchemaDao.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ private boolean isTrue(ResultSet resultSet, int index) {
         if (isPostgres()) {
             return Objects.equals(resultSet.getString(index), "YES");
         }
-        return resultSet.getBoolean(0);
+        return resultSet.getBoolean(index);
```
[the remark this answers is not in the document]
Comment (ShuranZhang) [reply src/main/java/io/debezium/connector/spanner/db/dao/SchemaDao.java]: Done.

File: src/main/java/io/debezium/connector/spanner/config/BaseSpannerConnectorConfig.java [hunk 2]
```diff
@@ -71,6 +71,7 @@ public abstract class BaseSpannerConnectorConfig extends CommonConnectorConfig {
     private static final String CONNECTOR_SPANNER_SYNC_SEGMENT_MS_POLICY_PROPERTY_NAME = "connector.spanner.sync.segment.ms";
     private static final String CONNECTOR_SPANNER_SYNC_MIN_CLEANABLE_DIRTY_RATIO_PROPERTY_NAME = "connector.spanner.sync.min.cleanable.dirty.ratio";
     private static final String CONNECTOR_SPANNER_SYNC_KAFKA_BOOTSTRAP_SERVERS_PROPERTY_NAME = "connector.spanner.sync.kafka.bootstrap.servers";
+    public static final String CONNECTOR_SPANNER_SYNC_TOPIC_INTERNAL_PREFIX = "connector.spanner.sync.topic.internal.";
```
[the remark this answers is not in the document]
Comment (ShuranZhang) [reply src/main/java/io/debezium/connector/spanner/config/BaseSpannerConnectorConfig.java]: Done.

File: src/main/java/io/debezium/connector/spanner/kafka/internal/KafkaInternalTopicAdminService.java [hunk 3]
```diff
@@ -64,6 +69,16 @@ public void createVerifySyncTopic() {
                 topicProps.put(TopicConfig.MIN_CLEANABLE_DIRTY_RATIO_CONFIG, config.syncMinCleanableDirtyRatio());
                 topicProps.put(TopicConfig.MAX_MESSAGE_BYTES_CONFIG, config.syncTopicMaxMessageSize());
 
+                Map<String, String> syncTopicInternalProps = config.getConfig().asMap().entrySet()
```
[the remark this answers is not in the document]
Comment (ShuranZhang) [reply src/main/java/io/debezium/connector/spanner/kafka/internal/KafkaInternalTopicAdminService.java]: Done.

**Comment by ShuranZhang** [comment]
> Please extend the scope of the issue to cover all topics and also all producer and consumer factories. It is not going to be a lot of code yet it will be very useful for the user.

I have added same feature for rebalancing topic. For all the producer and consumer factories, this feature is already included since all consumer and producer factories utilize [kafkaProps](https://github.com/debezium/debezium-connector-spanner/blob/d6dfb6fb761117142ba5af9e6af61f576273aec2/src/main/java/io/debezium/connector/spanner/SpannerConnectorConfig.java#L80), and all pass through configurations with prefix [KAFKA_INTERNAL_CLIENT_CONFIG_PREFIX](https://github.com/debezium/debezium-connector-spanner/blob/d6dfb6fb761117142ba5af9e6af61f576273aec2/src/main/java/io/debezium/connector/spanner/config/BaseSpannerConnectorConfig.java#L94) have been extracted to this kafkaProps. See this example usage form [ProducerFactory](https://github.com/debezium/debezium-connector-spanner/blob/main/src/main/java/io/debezium/connector/spanner/kafka/internal/ProducerFactory.java#L29).

**Review by jpechane — APPROVED** [review]
@ShuranZhang LGTM, thanks! Waiting for ready/test label before merging.
In the meantime could you please prepare a separate PR in the main repo with docs update?

**Comment by ShuranZhang** [comment]
> @ShuranZhang LGTM, thanks! Waiting for ready/test label before merging. In the meantime could you please prepare a separate PR in the main repo with docs update?

Added the label and will work on updating the doc with in a separate pr. Thanks!

**Comment by jpechane** [comment]
@ShuranZhang Applied, thanks!

**Pull request merged**